src/lsstseries/analysis/structure_function_calculators/structure_function_calculator.py

```diff
@@ -0,0 +1,93 @@
+from typing import List
+
+import numpy as np
+
+from lsstseries.analysis.structure_function_argument_containers.structure_function_argument_container import (
+    StructureFunctionArgumentContainer,
+)
+
+
+class StructureFunctionCalculator:
+    """This is the base class from which all other Structure Function calculator
+    methods inherit. Extend this class if you want to create a new Structure
+    Function calculation method.
+    """
+
+    def __init__(
+        self,
+        time: List[float],
+        flux: List[float],
+        err: List[float],
+        argument_container: StructureFunctionArgumentContainer,
+    ):
+        self._time = time
+        self._flux = flux
+        self._err = err
+        self._argument_container = argument_container
+
+        self._bins = argument_container.bins
+        self._binning_method = argument_container.bin_method
+        self._bin_count_target = argument_container.bin_count_target
+        self._dts = []
+        return
+
+    def calculate(self):
+        """Abstract method that must be implemented by the child class."""
+        raise (NotImplementedError, "Must be implemented by the child class")
+
+    def _bin_dts(self, dts):
```

Comment:
This method was taken from `structurefunction2.py`, and the only modifications here are to allow it to be used as a class method.